Session 기반 인증 JWT 기준으로 변경 및 테스트한다

voidmatcha opened the pull request:
> resolve #8
## Detail & Solution
서버의 인증방식이 세션에서 JWT로 변경되어 이를 적용할 필요성이 있었습니다.

## What I did
- serverContext에 axios 설정에서 헤더에 토큰을 포함시키도록 하였습니다.(sessionStorage에서 꺼내올 수 있게끔 처리)
- 기존 URL을 임시 서버 URL로 변경하였습니다.
- 동건님이 변경하신 엔드포인트에 맞춰 API request, response 수정 및 리포지토리 경로를 수정하였습니다.
- 기존에 로그인하면 사용자 정보를 response하는 구조에서 토큰만 response하는 구조로 변경되었으므로 이를 반영하였습니다.
  - 유저 정보를 가져올 수 있는 info API 요청용 리포지토리 추가하였습니다.
- auth 리듀서에 JWT 토큰을 추가하려 했으나 revert하고 sessionStorage에서만 관리하도록 하였습니다.
- API명세에 맞춰 스터디방 생성 Category dropdown을 추가하였습니다. https://github.com/witherview/witherview_frontend/pull/13/commits/1fd1255602fe34b5e97b6d4ec20450730eff5600
- promise로 Axios 요청하는 부분을 async await으로 변경하였습니다 https://github.com/witherview/witherview_frontend/pull/13/commits/db3fc6498a94aea8e631c4297189ec99fb44329e
- JWT 토큰을 기반으로 AuthRoute 에서 유저 정보를 가져오도록 하는 로직을 추가하였습니다. https://github.com/witherview/witherview_frontend/pull/13/commits/da0861a3d66ee2f7bf95783455d4ef7c3781b8e1
  - 새로고침 하는 경우 랜딩페이지로 리다이렉트되는것이 아니라 페이지에 머물 수 있도록 처리하였습니다.
- 서버 컨트롤러 구조에 맞춰 Axios Repository를 수정하였습니다. https://github.com/witherview/witherview_frontend/pull/13/commits/a514905ee31a0246763bdec9637bd2436ca799b3
  - 추가해야 할 Repository는 TODO로 주석을 달아놓았습니다.
  - TODO는 #48 을 통해 해결합니다.

## What I need to do
- [x] 새로 배포된 테스트 서버에 프로필 이미지 POST 쪽에 문제가 있는 것 전달 및 해결 후 테스트

voidmatcha committed:
feat: 🎸 api 리포지토리에 새로 생성된 엔드포인트 추가

- putProfileInfoApi: 프로필 정보 업데이트
- postProfileImageApi: 프로필 이미지 업데이트

voidmatcha committed:
feat: 🎸 프로필 이미지 수정 및 업로드 로직 추가

- authReducer에 image 추가
- ProfileEdit에 이미지 가져오는 부분 추가
- WelcomePage에 이미지 POST 로직 추가

voidmatcha committed:
feat: 🎸 닉네임 수정하는 로직 추가 (#4)

- 예은님이 로그인 response 수정하는 것을 고려해서 작업
- 관심산업 및 직무 태그에 put 써야해서 authReducer에 해당 스토어 추가
- 각각의 REST 메소드가 실행될 경우에만 실행되도록 함

voidmatcha committed:
refactor: 💡 eslint rule 추가

- no-return-await 옵션 추가
- import/prefer-default-export 옵션 추가

voidmatcha committed:
Merge branch 'feature/#11' into feature/#8

voidmatcha committed:
refactor: 💡 useInfo 데이터 상태 sessionStorage 및 AuthRoute에 추가

- 이렇게 해야 로그인을 다시하면 해당 정보들을 세션스토리지에서 꺼내올 수 있음

voidmatcha committed:
refactor: 💡 ProfileEdit 컴포넌트 이미지 선택시 POST하도록 변경

- 원래 상위컴포넌트인 WelcomePage의 버튼을 클릭했을 때 POST되던걸 ProfileEdit에서 선택시 POST하도록 함

voidmatcha committed:
fix: 🐛 redux auth store에 imageFile 상태 제거

- formData를 저장하는 형태였으나 해당 상태를 저장할 필요가 없어짐
- 상위컴포넌트에 상태를 보내는 용도로 사용했었음 근데 formData를 저장하면 시리얼라이저블 에러가남

voidmatcha committed:
refactor: 💡 redux auth store의 imgae 상태 처리하는 로직 추가

- 로그인 완료시 sessionStorage에 저장하도록 함
- 프로필 이미지 적용되는 부분 useSelector로 base64 이미지 가져오도록 함
- seLogin dispatch에 image도 포함되도록 하였으므로 이를 사용하는 부분  리펙토링

voidmatcha committed:
Merge branch 'feature/#4' into feature/#8

voidmatcha committed:
refactor: 💡 jwt 구조에 맞춰 요청시 헤더에 Authorization 추가한다 (#8)

- 상태를 가져올 수 있을지는 모르겠어서 sessionStorage에서 토큰값을 꺼내오도록 함
- 서버 주소 변경

voidmatcha committed:
refactor: 💡 바뀐 엔드포인트에 맞게 변경 및 getUserInfoApi 추가한다 (#8)

- groupRepository 및 userRepository에서 바뀌어야 할 부분 바꿈

voidmatcha committed:
refactor: 💡 jwt access 토큰 상태 auth 리듀서에 추가한다 (#8)

- setImageFile export 실수로 안지운 부분 지움
- isLogin은 setAccessToken에서 처리하도록 변경

voidmatcha committed:
refactor: 💡 로그인, 회원가입, AuthRoute를 jwt 구조에 맞춰 리펙토링한다 (#8)

- jwt에 맞춰 바뀐 구조에 따라 로그인, 회원가입 리팩토링 함
- AuthRoute도 리펙토링함

voidmatcha committed:
refactor: 💡 스터디방 생성에 Category dropdown 추가한다 (#8)

- 리뉴얼된 해당 엔트포인트 body에 추가되어 필요했음
- 추후에 필터링 및 검색의 방향성에 따라 바뀔 가능성이 있음

Sung-jin commented on src/components/molecules/ProfileEdit/ProfileEdit.js:
```diff
@@ -29,11 +33,45 @@ const IconWrapper = styled.span`
+    };
+    reader.onerror = (error) => {
+      console.error(error);
+    };
```
로그는 테스트 확인용 로그인가요?

불필요하다면 제거해도 좋을 것 같습니다.
저는 해보지 않아서 궁금한게요, catch 부분에 실패를 한다면 어떻게 처리 하시나요? 에러페이지로 가거나 오류 안내 모달 등을 띄우나요?

voidmatcha replied:
이부분은 모달로 처리해야 할 듯 합니다! 여기 외에도 alert로 처리되는 부분이 꽤 있는데 재사용성 고려해서 만들어볼게요

Sung-jin commented on src/pages/WelcomePage/WelcomePage.js:
```diff
@@ -138,27 +183,35 @@ export default function WelcomePage({ history }) {
-            개인정보처리방침
-          </WrapAnker>
+          <WrapAnker onClick={() => {}}>이용약관</WrapAnker>
+          <WrapAnker onClick={() => {}}>개인정보처리방침</WrapAnker>
```
이러한 빈 이벤트는 추후에 약관 파일이 생성되면 다운로드 기능을 연결하기 위해서 작성하신건가요?

TODO 등으로 구분을 해 주시면 나중에 지나치지 않을 것 같습니다.

voidmatcha replied:
네넵 TODO 주석 추가하도록 하겠습니다~

voidmatcha commented:
에러코드 제거  및 모달 추가와 주석은 별도의 Issue(#25) 및 PR로 처리하도록 하겠습니다.

voidmatcha commented:
response body의 accessToken 네이밍이 access_token으로 변경됨

voidmatcha committed:
refactor: 💡 Axios promise 형태를 async await 으로 변경한다 (#8)

- promise 구조를 async await 구조로 변경
- try catch로 예외처리

voidmatcha committed:
refactor: 💡 일부 API 요청시 변경된 body 의 key 를 반영한다 (#8)

- POST: /api/group/feeedback
  - id => studyRoomId
  - targetUser => recievedUser
  - studyHistoryId 추가

- POST: /api/group/room
  - time에서 초까지 포함하도록 변경

- GET: /api/self/queston
  - request param에서 /로 id를 전달하도록 변경

voidmatcha committed:
Revert "refactor: 💡 jwt access 토큰 상태 auth 리듀서에 추가"

- accessToken을 전역 상태로 관리할 필요는 없기 때문에 revert
This reverts commit 30952928c97feadad1f43e00b9d5e22c46dbc929.

voidmatcha committed:
refactor: 💡 auth 리듀서 login 및 logout 시 관리할 상태를 수정한다 (#8)

- isLogin을 payload 받을 수 있도록 setLogin 수정
- setLogout시 sessionStorage removeItem 실행되도록 함
  - 직무랑 분야 초기화 하도록 수정

voidmatcha committed:
refactor: 💡 로그인 및 회원가입 로직을 JWT에 맞춰 리팩토링 한다 (#8)

- 로그인
  - test7@test.com 로 기본 계정 변경
  - getUserInfoApi 및 dispatch 부분 AuthRoute에서 처리하도록 하여 이부분 제거
  - redirect는 sessionStorage에 accessToken 이 존재 할 경우에 하도록 수정

- 회원가입
  - promise를 async await으로
  - function을 arrow function으로
  - redirect 하는 부분 없앰

- 프로필 이미지 src, image, profileDefault 순으로 OR로 선택되도록 수정

voidmatcha committed:
refactor: 💡 AuthRoute 에서 redirect 조건 수정 및 유저 정보 요청 추가한다 (#8)

- redirect 조건은 로그인이 안되어있고 accessToken이 세션스토리지에 없을 경우
- isLogin 상태가 false이고 accesToken이 존재하는 경우 getUserInfoApi로 유저 정보 가져오도록 함
- App.js에 AuthRoute 사용하는 부분 수정

voidmatcha committed:
refactor: 💡 서버 컨트롤러에 맞춰 Axios Repository 수정 및 반영한다 (#8)

- 규칙없이 분리되어 있던 Axios Repository 서버 컨트롤러에 맞춰서 리팩토링
  - 구현 안되어있는 부분 TODO로 추가
- 수정된 Axios Repository에 맞춰 컴포넌트 코드 수정

✅ Closes: #8

Sung-jin reviewed (approved):
고생하셨습니다. 👍

then().catch() -> await 으로 변경하면서 try 블록 안에 console.log 가 가끔 보이는데, 추후 프로덕션으로 올릴때 로그관련해서는 전체적으로 정리하면 좋을 것 같습니다.

voidmatcha commented:
> 고생하셨습니다. 👍
> 
> then().catch() -> await 으로 변경하면서 try 블록 안에 console.log 가 가끔 보이는데, 추후 프로덕션으로 올릴때 로그관련해서는 전체적으로 정리하면 좋을 것 같습니다.

console.log 제거까지 반영하고 merge 하려고 했는데 이미 다른 PR에서 리팩토링 하면서 지우거나 구조상 수정된 코드들이 좀 있어서 전체적으로 머지되고 남아있으면 그때 정리해도록 하겠습니다!

voidmatcha committed:
Merge branch 'develop' into feature/#8

voidmatcha commented:
merge한 의존성 PR 브랜치가 develop에 merge 될 경우 해당 커밋들이 가려질 줄 알았는데 아니었네요. 다음부터는 이런 형태로 작업할 경우 리베이스 하도록 하겠습니다.
